sklearn/inspection/_permutation_importance.py

```diff
@@ -58,7 +58,7 @@ def _calculate_permutation_scores(
         if hasattr(X_permuted, "iloc"):
             col = X_permuted.iloc[shuffling_idx, col_idx]
             col.index = X_permuted.index
-            X_permuted.iloc[:, col_idx] = col
```

Comment:
This is a bit weird to use a slightly more complicated syntax just to avoid a warning but oh well I don't think there is anything better to do ...

Reply:
I am still thinking about how to permute the values of a dataframe column ignoring the indices.